src/megaapi_impl.cpp

```diff
@@ -7295,6 +7295,15 @@ void MegaApiImpl::request_error(error e)
         request->setText(client->sslfakeissuer.c_str());
     }
 
+    if (e == API_ESID)
+    {
+        if (client->sctable)
+        {
+            client->sctable->remove();
+        }
```

Comment:
I think that the cache of syncs should be also removed here like in `CommandLogout::procresult`